rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/TclLauncher.xtend

```diff
@@ -88,4 +100,30 @@ class TclLauncher implements Launcher {
 		return true
 	}
 
+	private def boolean launchMavenBasedTest(IStructuredSelection selection, IProject project, String elementId) {
+		logger.info("Trying to launch maven test execution")
+
+		val job = new Job("Execute test " + elementId) {
+
+			override protected run(IProgressMonitor monitor) {
+				val mvnExec = new MavenExecutor
```

Comment:
We should not use the new operator here.
Better: 

`@Inject Provider<MavenExecutor> executorProvider //javax.inject `
...
`val executor = executorProvider.get`